src/topology/algebra/uniform_group.lean

```diff
@@ -400,6 +412,14 @@ local attribute [instance] topological_group.to_uniform_space
 @[to_additive] lemma uniformity_eq_comap_nhds_one' :
   𝓤 G = comap (λp:G×G, p.2 / p.1) (𝓝 (1 : G)) := rfl
 
+@[to_additive] lemma topological_group_is_uniform_of_compact_space
+  [compact_space G] [t2_space G] : uniform_group G :=
+⟨begin
+  haveI : separated_space G := separated_iff_t2.mpr (by apply_instance),
+  apply compact_space.uniform_continuous_of_continuous,
+  exact continuous_div',
+end⟩
+
```

Comment:
Maybe mention that this is true for any topological group where the left and right uniformities agree, and that both this and `topological_comm_group_is_uniform` should eventually become consequence of this general fact?

Reply:
You're making great points but I'm also a bit reluctant to say this given that we don't have a convenient means of expressing this using our existing typeclasses.